Flaky test: Test/CZTraceOverwriteChannelDeletion

```
--- FAIL: Test (61.15s)
    --- FAIL: Test/CZTraceOverwriteChannelDeletion (10.19s)
    	tlogger.go:97: clientconn.go:164: Channel Created
    	tlogger.go:97: clientconn.go:242: parsed scheme: "c0pjmx7gnfxw"
    	tlogger.go:97: resolver_conn_wrapper.go:143: ccResolverWrapper: sending update to cc: {[{127.0.0.1:0 grpclb.server <nil> 1 <nil>}] <nil> <nil>}
    	tlogger.go:97: resolver_conn_wrapper.go:218: Resolver state updated: {Addresses:[{Addr:127.0.0.1:0 ServerName:grpclb.server Attributes:<nil> Type:1 Metadata:<nil>}] ServiceConfig:<nil> Attributes:<nil>} (resolver returned new addresses)
    	tlogger.go:97: clientconn.go:666: ClientConn switching balancer to "grpclb"
    	tlogger.go:97: clientconn.go:681: Channel switches to new LB policy "grpclb"
    	tlogger.go:97: clientconn.go:154: Channel Created
    	tlogger.go:97: clientconn.go:154: Nested Channel(id:2) created
    	tlogger.go:97: clientconn.go:242: parsed scheme: "grpclb-internal"
    	tlogger.go:97: resolver_conn_wrapper.go:143: ccResolverWrapper: sending update to cc: {[{127.0.0.1:0 grpclb.server <nil> 0 <nil>}] <nil> <nil>}
    	tlogger.go:97: resolver_conn_wrapper.go:218: Resolver state updated: {Addresses:[{Addr:127.0.0.1:0 ServerName:grpclb.server Attributes:<nil> Type:0 Metadata:<nil>}] ServiceConfig:<nil> Attributes:<nil>} (resolver returned new addresses)
    	tlogger.go:97: clientconn.go:666: ClientConn switching balancer to "pick_first"
    	tlogger.go:97: clientconn.go:681: Channel switches to new LB policy "pick_first"
    	tlogger.go:97: clientconn.go:722: Subchannel Created
    	tlogger.go:97: clientconn.go:722: Subchannel(id:3) created
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:416: Channel Connectivity change to CONNECTING
    	tlogger.go:97: resolver_conn_wrapper.go:143: ccResolverWrapper: sending update to cc: {[{127.0.0.1:0  <nil> 0 <nil>}] 0xc420baef00 <nil>}
    	tlogger.go:97: resolver_conn_wrapper.go:218: Resolver state updated: {Addresses:[{Addr:127.0.0.1:0 ServerName: Attributes:<nil> Type:0 Metadata:<nil>}] ServiceConfig:0xc420baef00 Attributes:<nil>} (service config updated)
    	tlogger.go:97: clientconn.go:666: ClientConn switching balancer to "round_robin"
    	tlogger.go:97: clientconn.go:681: Channel switches to new LB policy "round_robin"
    	tlogger.go:97: clientconn.go:722: Subchannel Created
    	tlogger.go:97: clientconn.go:722: Subchannel(id:4) created
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:416: Channel Connectivity change to SHUTDOWN
    	tlogger.go:97: clientconn.go:416: Channel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to SHUTDOWN
    	tlogger.go:97: clientconn.go:1426: Subchannel Deleted
    	tlogger.go:97: clientconn.go:1426: Subchanel(id:3) deleted
    	tlogger.go:97: clientconn.go:1011: Channel Deleted
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0  <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0 grpclb.server <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to TRANSIENT_FAILURE
    	tlogger.go:97: clientconn.go:416: Channel Connectivity change to TRANSIENT_FAILURE
    	tlogger.go:97: clientconn.go:1011: Nested channel(id:2) deleted
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0  <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to TRANSIENT_FAILURE
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0  <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to TRANSIENT_FAILURE
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0  <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to TRANSIENT_FAILURE
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to CONNECTING
    	tlogger.go:97: clientconn.go:1192: Subchannel picks a new address "127.0.0.1:0" to connect
    	tlogger.go:97: clientconn.go:1274: grpc: addrConn.createTransport failed to connect to {127.0.0.1:0  <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0: getsockopt: connection refused". Reconnecting...
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to TRANSIENT_FAILURE
    	channelz_test.go:1936: nested channel should have been deleted since its parent's trace should not contain any reference to it anymore
    	tlogger.go:97: clientconn.go:416: Channel Connectivity change to SHUTDOWN
    	tlogger.go:97: clientconn.go:1055: Subchannel Connectivity change to SHUTDOWN
    	tlogger.go:97: clientconn.go:1426: Subchannel Deleted
    	tlogger.go:97: clientconn.go:1426: Subchanel(id:4) deleted
    	tlogger.go:97: clientconn.go:1011: Channel Deleted
    	tlogger.go:111: types.go:69: attempt to delete child with id 4 from a parent (id=1) that doesn't currently exist
FAIL
FAIL	google.golang.org/grpc/test	140.850s
```

--- a/test/channelz_test.go
+++ b/test/channelz_test.go
@@ -1925,6 +1925,10 @@ func (s) TestCZTraceOverwriteChannelDeletion(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// If nested channel deletion is last trace event before the next validation, it will fail, as the top channel will hold a reference to it.
+	// This line forces a trace event on the top channel in that case.
+	r.UpdateState(resolver.State{Addresses: []resolver.Address{{Addr: "127.0.0.1:0"}}, ServiceConfig: parseCfg(r, `{"loadBalancingPolicy": "round_robin"}`)})
+
 	// verify that the nested channel no longer exist due to trace referencing it got overwritten.
 	if err := verifyResultWithDelay(func() (bool, error) {
 		cm := channelz.GetChannel(nestedConn)

